analyzers/src/SonarAnalyzer.Common/Rules/SymbolicExecutionRunnerBase.cs

```diff
@@ -22,123 +22,142 @@
+{
+    protected abstract ImmutableDictionary<DiagnosticDescriptor, RuleFactory> AllRules { get; }
+    protected abstract ControlFlowGraph CreateCfg(SemanticModel model, SyntaxNode node, CancellationToken cancel);
+    protected abstract void AnalyzeSonar(SonarSyntaxNodeReportingContext context, SyntaxNode body, ISymbol symbol);
 
-        private protected /* for testing */ SymbolicExecutionRunnerBase(IAnalyzerConfiguration configuration) =>
-            Configuration = configuration;
+    protected IAnalyzerConfiguration Configuration { get; }
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => AllRules.Keys.ToImmutableArray();
-        protected override bool EnableConcurrentExecution => false;
+    private protected /* for testing */ SymbolicExecutionRunnerBase(IAnalyzerConfiguration configuration) =>
+        Configuration = configuration;
 
-        protected static RuleFactory CreateFactory<TRuleCheck>() where TRuleCheck : SymbolicRuleCheck, new() =>
-            new RuleFactory<TRuleCheck>();
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => AllRules.Keys.ToImmutableArray();
+    protected override bool EnableConcurrentExecution => false;
 
-        protected static RuleFactory CreateFactory<TRuleCheck, TSonarFallback>()
-            where TRuleCheck : SymbolicRuleCheck, new()
-            where TSonarFallback : new() =>
-            new RuleFactory<TRuleCheck, TSonarFallback>();
+    protected static RuleFactory CreateFactory<TRuleCheck>() where TRuleCheck : SymbolicRuleCheck, new() =>
+        new RuleFactory<TRuleCheck>();
 
-        // We need to rewrite this https://github.com/SonarSource/sonar-dotnet/issues/4824
-        protected static bool IsEnabled(SonarSyntaxNodeReportingContext context, DiagnosticDescriptor descriptor) =>
-            context.HasMatchingScope(descriptor) && descriptor.GetEffectiveSeverity(context.Compilation.Options) != ReportDiagnostic.Suppress;
+    protected static RuleFactory CreateFactory<TRuleCheck, TSonarFallback>()
+        where TRuleCheck : SymbolicRuleCheck, new()
+        where TSonarFallback : new() =>
+        new RuleFactory<TRuleCheck, TSonarFallback>();
 
-        protected void Analyze<TNode>(SonarAnalysisContext analysisContext, SonarSyntaxNodeReportingContext context, Func<TNode, SyntaxNode> getBody) where TNode : SyntaxNode
+    protected static bool IsEnabled(SonarSyntaxNodeReportingContext context, DiagnosticDescriptor descriptor)
+    {
+        if (context.HasMatchingScope(descriptor))
         {
-            if (getBody((TNode)context.Node) is { } body && context.SemanticModel.GetDeclaredSymbol(context.Node) is { } symbol)
+            // This is a reproduction of Roslyn activation logic
+            // https://github.com/dotnet/roslyn/blob/0368609e1467563247e9b5e4e3fe8bff533d59b6/src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs#L1316-L1327
```

Comment:
Exmplain the why. Maybe like so.
```suggestion
            // SE is called if _any_ SE rule is activated. We need to figure out if _this_ SE rule is activated.
            // This is a reproduction of Roslyn activation logic
            // https://github.com/dotnet/roslyn/blob/0368609e1467563247e9b5e4e3fe8bff533d59b6/src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs#L1316-L1327
```